github/azure-self-hosted-runners/tf/main.tf

```diff
@@ -0,0 +1,170 @@
+resource "azurerm_resource_group" "garm_rg" {
+  name     = var.resource_group_name
+  location = var.location
+}
+
+resource "azurerm_user_assigned_identity" "garm_id" {
+  name                = "garm"
+  location            = azurerm_resource_group.garm_rg.location
+  resource_group_name = azurerm_resource_group.garm_rg.name
+}
+
+data "azurerm_subscription" "current" {
+}
+
+data "azurerm_client_config" "current" {
+}
+
+resource "random_string" "suffix" {
+  length  = 6
+  lower   = true
+  upper   = false
+  special = false
+}
+
+resource "azurerm_role_assignment" "garm_role_as" {
+  scope                = data.azurerm_subscription.current.id
+  role_definition_name = "Contributor"
```

Comment:
These permissions are too wide open

Reply:
we cannot scope it to a resource group, since garm will create resource groups for instances